### PR TITLE
socat: add apparmor profile

### PR DESCRIPTION
**Pull request by nishitm**
Creating a profile for "socat". I have tested this profile on "socat" latest upstream version: 1.8.0.1 with its test suite on kernel `6.8.0-31`. I have written this profile with the latest abi 4.0 support. 

Let me know if it fails any tests or needs more modifications. Thanks

### CODE REVIEW

File: apparmor.d/groups/network/socat [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  file rw /dev/tty,
+
+  # Allow reading /dev/vsock
+  file r /dev/vsock,
+
+  # certs/keys can be are stored in:
+  #     - /etc/ssl/certs/*.{key,crt}
+  #     - $HOME/.cert/**/*.pem
+  file r /etc/ssl/certs/{,*.{key,crt}},
+  file r @{HOME}/.certs/{,**},
```
Comment (roddhjav) [inline apparmor.d/groups/network/socat:55]: Use the `ssl_certs` abstraction instead.

**Comment by roddhjav** [comment]
Thanks for this PR.

I would encourage you to read the profile guideline: https://apparmor.pujol.io/development/#add-a-profile and to adapt the profile accordingly.

```
  # Allow reading from /dev/tty
  file rw /dev/tty,
```
While it is a valid apparmor rule, we use the more common form:   `/dev/tty rw,` (btw, this comment is useless).

> I have written this profile with the latest abi 4.0 support.

You don't use abi 4 specific feature, so keep it in abi 3. I will all profiles at once when we move to abi 4 by default. (ABI 4 is already supported in distribution that support it).

File: apparmor.d/groups/network/socat [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  capability net_bind_service,
+  capability sys_resource,
+
+  # Allow creation of network sockets and `socat` uses dccp for some
+  # fuctionalities that is why it is necessary to allow whole `network`
+  network,
+
+  # Enale /dev/ptmx access for testsuite
+  # file rw /dev/ptmx,
+  # file rw /dev/pts/*,
```
Comment (roddhjav) [inline apparmor.d/groups/network/socat:36]: Use the `consoles` abstraction. (same for /dev/tty).

File: apparmor.d/groups/network/socat [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  # TUN/TAP device
+  file rw /dev/net/tun,
+
+  # Process-specific access
+  file rw @{PROC}/@{pid}/fd/*,
```
Comment (roddhjav) [inline apparmor.d/groups/network/socat:42]: Should be: `@{PROC}/@{pid}/fdinfo/@{int} rw,`

**Comment by nishitm** [comment]
thanks for the valuable feedback, I have updated the profile in https://github.com/roddhjav/apparmor.d/pull/454/commits/b9eb78333806580e15b3b4c2252c7b4887082e22

File: apparmor.d/groups/network/socat [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2024 Nishit Majithia (nishitm)
+# SPDX-License-Identifier: GPL-2.0-only
+# vim: ft=apparmor
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = /usr/bin/socat
```
Comment (roddhjav) [inline apparmor.d/groups/network/socat:10]: It must be: `@{bin}/socat`

File: apparmor.d/groups/network/socat [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  capability sys_admin,
+  capability fsetid,
+  capability chown,
+  capability net_bind_service,
+  capability sys_resource,
+
+  # Allow creation of network sockets and `socat` uses dccp for some
+  # fuctionalities that is why it is necessary to allow whole `network`
+  network,
+
```
Comment (roddhjav) [inline apparmor.d/groups/network/socat:31]: You need to insert the rule `@{exec_path} mr,`

Comment (nishitm) [reply apparmor.d/groups/network/socat:31]: Change has been added

**Comment by nishitm** [comment]
aah, I missed those pieces 
but it has been added. Thanks for pointing this out

**Comment by roddhjav** [comment]
Good, merged.

File: apparmor.d/groups/network/socat [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  # Enale /dev/ptmx access for testsuite
+  # /dev/ptmx rw,
+
+  # TUN/TAP device
+  /dev/net/tun rw,
+
+  # Process-specific access
+  @{PROC}/@{pid}/fdinfo/@{int} rw,
+  @{PROC}/@{pid}/stat           r,
```
Comment (nobody43) [inline apparmor.d/groups/network/socat:42]: @nishitm 
Not possible to restrict with `owner`?

**Pull request merged**